src/NuGet.Core/NuGet.Commands/ClientCertificatesCommand/ListClientCertRunner.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using NuGet.Common;
+using NuGet.Configuration;
+
+namespace NuGet.Commands
+{
+    public class ListClientCertRunner
+    {
+        public static void Run(ListClientCertArgs args, Func<ILogger> getLogger)
+        {
+            args.Validate();
+            var settings = RunnerHelper.GetSettings(args.Configfile);
+            var clientCertificateProvider = new ClientCertificateProvider(settings);
+
+            var items = clientCertificateProvider.GetClientCertificates();
+
+            if (!items.Any())
+            {
+                getLogger().LogInformation(Strings.NoClientCertificates);
+                return;
+            }
+
+            var clientCertificatesLogs = new List<LogMessage>();
+
+            getLogger().LogInformation(Strings.RegsiteredClientCertificates);
+            getLogger().LogInformation(Environment.NewLine);
+
+            for (var i = 0; i < items.Count; i++)
+            {
+                var item = items[i];
+
+                var builder = new StringBuilder();
+
+                var index = $" {i + 1,-6}.";
+                const string defaultIndentation = "      ";
+
+                builder.Append(index);
+                builder.AppendFormat(CultureInfo.CurrentCulture, Strings.ClientCertificatesLogTitle, item.PackageSource, item.ElementName);
+                builder.AppendLine();
```

Comment:
Update the string resource to use a placeholder for the numbered item with padding.

```suggestion
                builder.AppendLine(CultureInfo.CurrentCulture, Strings.ClientCertificatesLogTitle, index, item.PackageSource, item.ElementName);
```

Reply:
Guys you create problems out of nowhere :) In favor of less string concatenation with string.format you adding bunch of additional variables and not readable code. This code is not critical for performance and often used single time in one process. In fact this code will be called 1 time per command line process. So I will reformat this again into more readable form.